libkbfs/folder_branch_ops.go

```diff
@@ -1206,6 +1206,15 @@ func (fbo *folderBranchOps) initMDLocked(
 		return err
 	}
 
+	fbo.headLock.Lock(lState)
+	defer fbo.headLock.Unlock(lState)
+	// Some other thread got here first, so give up and let it go
+	// before we push anything to the servers.
```

Comment:
Yes, and they execute one after the other -- but each one assumes it is the first one to run this function.  When they call `MDOps.Put()`, the second two fail with a conflict because the first one already succeeded.